src/instructlab/sdg/utils/docprocessor.py

```diff
@@ -0,0 +1,595 @@
+        {filepaths=},
+        {documents=},
+    """)
+
+    # Check for single document
+    if isinstance(documents, str):
+        documents = [documents]
+        logger.info(
+            "Converted single string into list of strings. Assumed the string passed in is the document. Normally, chunk_document() should take a list as input."
+        )
+
+    assert type(documents) == list
+
+    md_docs, pdf_docs = _split_docs_by_filetype(zip(documents, filepaths))
+
+    chunked_mds = chunk_markdowns(md_docs, server_ctx_size, chunk_word_count)
+    chunked_pdfs = chunk_pdfs(pdf_docs, filepaths)
+
+    return chunked_mds + chunked_pdfs
+
+
+def chunk_markdowns(
+    documents: List | str, server_ctx_size, chunk_word_count
+) -> List[str]:
+    """Naively chunk markdown documents based on the word count provided by the user.
+    Args:
+        documents (list): List of markdown documents.
+        server_ctx_size (int): Context window size of server.
+        chunk_word_count (int): Maximum number of words to chunk a document.
+    Returns:
+         List[str]: List of chunked documents.
+    """
+    num_tokens_per_doc = _num_tokens_from_words(chunk_word_count)
+    if num_tokens_per_doc > int(server_ctx_size - 1024):
+        raise ValueError(
+            "Error: {}".format(
+                str(
+                    f"Given word count ({chunk_word_count}) per doc will exceed the server context window size ({server_ctx_size})"
+                )
+            )
+        )
+    if documents == []:
+        return []
+
+    # Placeholder for params
+    content = []
+    chunk_size = _num_chars_from_tokens(num_tokens_per_doc)
+    chunk_overlap = _DEFAULT_CHUNK_OVERLAP
+
+    # Using Markdown as default, document-specific chunking will be implemented in separate pr.
+    md_text_splitter = RecursiveCharacterTextSplitter.from_language(
+        language=Language.MARKDOWN,
+        chunk_size=chunk_size,
+        chunk_overlap=chunk_overlap,
+    )
+
+    # Determine file type for heuristics, default with markdown
+    for doc in documents:
+        # Use regex to remove unnecessary dashes in front of pipe characters in a markdown table.
+        doc = re.sub(r"-{2,}\|", "-|", doc)
+        # Remove unnecessary spaces in front of pipe characters in a markdown table.
+        doc = re.sub(r"\  +\|", " |", doc)
+        temp = md_text_splitter.create_documents([doc])
+        content.extend([item.page_content for item in temp])
+    return content
+
+
+def chunk_pdfs(pdf_docs: List, leaf_node_path: Path):
+    """Semantically chunk PDF documents.
+
+    TODO
+    """
+    if pdf_docs == []:
+        return []
+
+    print(f"""THIS IS KHALED: CHUNKING PDF DOCS
+        {pdf_docs[0]=}
+    """)
+
+    converter = DocumentConverter(pipeline_options=PipelineOptions())
+    parsed_pdfs = [converter.convert_single(d) for d in pdf_docs]
+    parsed_dicts = [p.render_as_dict() for p in parsed_pdfs]
+
+    docling_jsons_path = DOC_FILEPATH / "docling-jsons"
```

Comment:
Prefer `os.path.join` to combine paths to string concatenation - `os.path.join` handles various edge cases like ensuring we don't add double slashes and more complex paths.